components/camel-jpa/src/main/java/org/apache/camel/component/jpa/JpaPollingConsumer.java

```diff
@@ -44,7 +44,7 @@ public class JpaPollingConsumer extends PollingConsumerSupport {
 
     private static final Logger LOG = LoggerFactory.getLogger(JpaPollingConsumer.class);
 
-    private transient ExecutorService executorService;
+    private ExecutorService executorService;
```

Comment:
why remove `transient` here? Is there an issue sonar reported?

Reply:
Yes, there are. [If class is not `Serializable` then dont need to use transient](https://sonarcloud.io/project/issues?issues=AX4xQtpX2wn0z3gD78Js&open=AX4xQtpX2wn0z3gD78Js&id=apache_camel)

Reply:
Hmm, it is interesting! I wonder if the modifier should be `volatile` since there might be multi polling consumers? @orpiske wdyt?

Reply:
Yes correct it should be volatile

Reply:
Should I change this?

Reply:
Yes pleaase

Reply:
This will introduce another sonar pointing as [`volatile` should be used only with primitive fields](https://rules.sonarsource.com/java/RSPEC-3077).

Reply:
I think we could ignore this issue as `volatile` here is used to avoiding creating multi `ExecutorService` in `receive(long timeout)`.

Reply:
> I think we could ignore this issue as `volatile` here is used to avoiding creating multi `ExecutorService` in `receive(long timeout)`.

I don't think volatile is enough to protect the code in this particular scenario. A synchronized block / lock would be the correct to ensure this. AFAIK, it only guarantees the visibility updates of itself (and other variables declared around it) but cannot protect against concurrent writes of the variable. And, since we have 2 operations happening on that variable in that particular block (a fetch - from memory - and a write - to memory) it would still be entirely possible for 2 threads to concurrently fetch a null value at the same time.  

```
// need to use a thread pool to perform the task so we can support timeout
if (executorService == null) {
    executorService = getEndpoint().getComponent().getOrCreatePollingConsumerExecutorService();
}
```

That said ... I have seen this pattern used elsewhere in the code and I always wanted to fix it, but given a lot of other things we should take into consideration before fixing it, I've always left it aside as suggested fix for a major version. 

So, TL,DR: IMHO, `volatile` is not OK, but we can leave it be for a larger fix in the future.

Reply:
I agree and is there CAMEL issue for tracking these fixes?

Reply:
I am not aware of any ticket tracking this item in particular. However, it is [listed on our SonarCloud instance](https://sonarcloud.io/project/issues?resolved=false&rules=java%3AS3077&id=apache_camel).

Please, feel free to create a Jira ticket if you think we should track it.

Reply:
@orpiske `volatile` added as with talked about